geti_sdk/deployment/inference_hook_interfaces.py

```diff
@@ -0,0 +1,263 @@
+        Return string representation of the PostInferenceTrigger
+        """
+        return f"{type(self).__name__}({self._repr_info_})"
+
+    @classmethod
+    def from_dict(cls, input_dict: Dict[str, Any]) -> "PostInferenceTrigger":
+        """
+        Construct a PostInferenceTrigger from an input dictionary `input_dict`
+
+        :param input_dict: Dictionary representation of the PostInferenceTrigger
+        :return: Instantiated PostInferenceTrigger, according to the input dictionary
+        """
+        available_triggers = {
+            subcls.__name__: subcls for subcls in cls.__subclasses__()
+        }
+        triggers: List["PostInferenceTrigger"] = []
+        for trigger_name, trigger_args in input_dict.items():
+            target_trigger = available_triggers[trigger_name]
+            if target_trigger._override_from_dict_:
+                triggers.append(target_trigger.from_dict(trigger_args))
+            else:
+                triggers.append(target_trigger(**trigger_args))
+        return triggers[0]
+
+
+class PostInferenceAction(PostInferenceObject):
+    """
+    Base class for post inference actions. These are actions that are used in inference
+    hooks, and can be (conditionally) executed after inference
+
+    :param log_level: Log level to use in the action, current options are `info` or
+        `debug`. Defaults to `debug`
+    """
+
+    def __init__(self, log_level: str = "debug"):
+        super().__init__()
+        if log_level.lower() == "debug":
+            self.log_function = logging.debug
+        elif log_level.lower() == "info":
+            self.log_function = logging.info
+        else:
+            raise ValueError(
+                f"Unsupported log_level `{log_level}`, options are 'info' or 'debug'."
+            )
+        self._repr_info_: str = ""
+
+    @abstractmethod
+    def __call__(
+        self,
+        image: np.ndarray,
+        prediction: Prediction,
+        score: Optional[float] = None,
+        name: Optional[str] = None,
+        timestamp: Optional[datetime.datetime] = None,
+    ):
+        """
+        Execute the action for the given `image` with corresponding `prediction` and an
+        `score`. The `score` could be passed from the trigger that triggers the action.
+
+        :param image: Numpy array representing an image
+        :param prediction: Prediction object which was generated for the image
+        :param score: Optional score computed from a post inference trigger
+        :param name: String containing the name of the image
+        :param timestamp: Datetime object containing the timestamp belonging to the
+            image
+        """
+        return NotImplemented
+
+    def __repr__(self) -> str:
+        """
+        Return string representation of the PostInferenceAction
+        """
+        return f"{type(self).__name__}({self._repr_info_})"
+
+    @classmethod
+    def from_dict(cls, input_dict: Dict[str, Any]) -> "PostInferenceAction":
```

Comment:
This method looks the same as in `Trigger` interface, could we push it to the base class?